### PR TITLE
Enable multi-label image classification in pipeline

### LINKED ISSUE #429: change activation function in Inference API for Image Classification
Hi all,
I have trained a multilabel Image Classification model here :
[https://huggingface.co/lombardata/dino-base-2023_10_31-demo-v5
](url)
and set the task in the model card to **image classification**. This allows the automatic creation of an Inference API on the model page, but the default activation function is a Softmax.
I would like to change the activation function to a Sigmoid one (the standard activation function for Multilabel Image Classification problems), but if I try to change the `pipeline_tag` in the` Model Card`, there is no option for Multilabel Image Classification in the Computer Vision task.
Do you have any suggestions on how to do it?
Thank you in advance!

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Enables multilabel image classification in the pipeline and explicity specifying the activation function applied to the model's logits - matching the logic for [text classification pipeline](https://github.com/huggingface/transformers/blob/ffd3710391c0700a3957f0cdf2c99bc5ae966c70/src/transformers/pipelines/text_classification.py#L195).

Fixes https://github.com/huggingface/huggingface.js/issues/429

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by amyeroberts** [comment]
@Narsil  - I've added applying sigmoid function to the model outputs for image classification matching the text classification pipeline, per your [suggestion here](https://github.com/huggingface/huggingface.js/issues/429#issuecomment-1880625080). 

Agree controlling through a config would be a good idea -- it is partially controlled through `model.config.problem_type` here. Is this enough or would you like the activation function to be more explicitly defined in the config?

**Review by Narsil — APPROVED** [review]
LGTM thanks for adding this.

**Review by ArthurZucker — APPROVED** [review]
LGTM 🤗
